plugin/build/ios/appDelegate.js

```diff
@@ -0,0 +1,85 @@
+        // https://regex101.com/r/nHrTa9/1/
+        // if the above regex fails, we can use this one as a fallback:
+        const fallbackInvocationLineMatcher = /-\s*\(BOOL\)\s*application:\s*\(UIApplication\s*\*\s*\)\s*\w+\s+didFinishLaunchingWithOptions:/g;
+        if (!modResults.contents.includes("#import <PushKit/PushKit.h>")) {
+            modResults.contents = modResults.contents.replace(/#import "AppDelegate.h"/g, `#import "AppDelegate.h"
+#import <PushKit/PushKit.h>
+#import "RNVoipPushNotificationManager.h"
+#import "RNCallKeep.h"`);
+        }
+        try {
+            modResults.contents = (0, generateCode_1.mergeContents)({
+                tag: "RNVoipPushNotificationAppDelegate",
+                src: modResults.contents,
+                anchor: methodInvocationLineMatcher,
+                offset: 0,
+                comment: "// ",
+                newSrc: methodInvocationBlock,
+            }).contents;
+        }
+        catch (e) {
+            // Fallback to the other regex
+            modResults.contents = (0, generateCode_1.mergeContents)({
+                tag: "RNVoipPushNotificationAppDelegate",
+                src: modResults.contents,
+                anchor: fallbackInvocationLineMatcher,
+                offset: 0,
+                comment: "// ",
+                newSrc: methodInvocationBlock,
+            }).contents;
+        }
+        // Add PushKit delegate method to the bottom of the file
+        // if other appDelegates are being implemented I will need to add this to the bottom of the file
+        if (!modResults.contents.includes("/* Add PushKit delegate method */")) {
+            modResults.contents = modResults.contents.replace(/@end/g, `/* Add PushKit delegate method */
+- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(PKPushType)type
+{
+    [RNVoipPushNotificationManager didUpdatePushCredentials:credentials forType:(NSString *)type];
+}
+
+- (void)pushRegistry:(PKPushRegistry *)registry didInvalidatePushTokenForType:(PKPushType)type
+{
+
+}
+
+- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(PKPushType)type withCompletionHandler:(void (^)(void))completion
+{
+    NSString *uuid = payload.dictionaryPayload[@"uuid"];
+    NSString *callerName = [NSString stringWithFormat:@"%@ (Connecting...)", payload.dictionaryPayload[@"callerName"]];
+    NSString *handle = payload.dictionaryPayload[@"handle"];
+
+    [RNVoipPushNotificationManager addCompletionHandler:uuid completionHandler:completion];
+
+    [RNVoipPushNotificationManager didReceiveIncomingPushWithPayload:payload forType:(NSString *)type];
+
+    [RNCallKeep reportNewIncomingCall: uuid
```

Comment:
should also grab `hasVideo` and maybe nested `payload` from `payload.dictionaryPayload`

`hasVideo` is especially important because iOS presents a different incoming call message label to inform user of type of call

Reply:
I have the isVideo payload to handle this.